lib/diagnostic.js

```diff
@@ -47,7 +53,9 @@ let scopes = [/* your response here */];
 // executing it.
 //
 let fireOnAlderaan = function () {
-  // your code here
+  deathStar.fire.this;
+  // this.What?.Iamnotgettingit.blowupalderaan(ItSucks);
+  // return this.argh;
```

Comment:
a few things here:
1) you want this function to `return` something. Right now, it's not returning anything
2) you instinct to use `this` is correct, but the point here is to define what the `this` in line 46 is. And in this case, we that value to be `alderaan`
3) the method you need to use to make this happen is `.call`, like this:

```js
let fireOnAlderaan = function () {
  return deathStar.fire.call(alderaan);
};
```